core/aws_ddk_core/stages/s3_event.py

```diff
@@ -33,6 +33,7 @@ def __init__(
         event_names: List[str],
         bucket_name: str,
         key_prefix: Optional[str] = None,
+        key_prefixes: Optional[List[str]] = None,
```

Comment:
Do we want to have two parameters for this? We could instead make `key_prefix` of type `Optional[Union[str, List[str]]]`. Then, we'd check if `key_prefix` was a single object, and if so, turn it into a singleton list.

Reply:
Kind of how we do for S3 paths in many wrangler functions

Reply:
The only reason I left it intact was to avoid a breaking change. Are we ok with swapping this out for a minor version release? @kukushking @LeonLuttenberger

Reply:
In my head I thought you meant changing the arg name to plural.

That makes sense, I'll just go with the List + str support for `key_prefix`